arelle/ValidateXbrlCalcs.py

```diff
@@ -694,58 +695,133 @@ def rangeValue(value, decimals=None, truncate=False) -> tuple[decimal.Decimal, d
                 return (vDecimal - dd, vDecimal + dd, False, False)
     return (vDecimal, vDecimal, True, True)
 
-def insignificantDigits(value, precision=None, decimals=None, scale=None) -> tuple[Decimal, Decimal] | None:
+
+def finiteDecimalsInt(decimals: int | float | str | None) -> int | None:
+    """
+    Given a decimals value, returns a finite integer
+    representation of the value, if possible. Otherwise, returns None.
+    :param decimals: Numeric fact decimals attribute value.
+    :return: Finite integer representation of given value, otherwise None
+    """
+    if isinstance(decimals, int):
+        return decimals
+    if decimals is None or decimals == "INF":
+        return None
+    if isinstance(decimals, float) and (isnan(decimals) or isinf(decimals)):
+        return None
     try:
-        vDecimal = decimal.Decimal(value)
-        if scale:
-            iScale = int(scale)
-            vDecimal = vDecimal.scaleb(iScale)
-        if precision is not None:
-            vFloat = float(value)
-            if scale:
-                vFloat = pow(vFloat, iScale)
-    except (decimal.InvalidOperation, ValueError): # would have been a schema error reported earlier
+        return int(decimals)
+    except ValueError:  # would be a schema error
```

Comment:
FWIW, I know these comments are pre-existing, but I don't know what they mean.  `decimals` is `decimalsType`, which includes `xs:integer`, which has no bound.  Of course, we can't/don't support arbitrarily large values...

Reply:
`ModelFact.decimals` is a `str`, and in most cases it is passed directly into `insignificantDigits`. I think the try/except here is intended to safely handle str values that may not be valid integers at all (which would have already caused a schema error), rather than concern over arbitrarily large values. I'd agree with the suggestion that we should probably be handling that case sooner in the process, but this is where I've drawn the line for the scope of this PR.